src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -261,10 +262,10 @@ class Namer { typer: Typer =>
         ctx.error(msg, tree.pos)
         name.freshened
       }
-      def preExisting = ctx.effectiveScope.lookup(name)
+      lazy val preExisting = ctx.effectiveScope.lookup(name)
```

Comment:
I don't think we need a lazy val here. preExisting is only used once on each path. Note that defs are more compact (in terms of byte code length) and efficient than lazy vals.